routes/index.js

```diff
@@ -1,75 +1,89 @@
+			ignorePunctuation: true
+		}));
+		
 		let renderData = {
 			path: 'none',
-			students: activeStudents
+			students: activeStudents,
+			registrations: activeRegistrations,
+			titles: await module.exports.getProgramTitles(activeStudents, activeRegistrations),
+			grades: constants.getGradeLevels(),
 		}
 
-		response.render('index', renderData);
+		response.render('index', renderData)
 		
 	},
 
 	sortFirstNames: async function(request, response) {
 		let studentList = await db.getStudentsList();
-		let activeStudents = [];
-		for (let i = 0; i < studentList.length; i++) {
-			if (studentList[i].status == "active") {
-				activeStudents.push(studentList[i]);
-			}
-		}
+		let activeStudents = programFile.activeStudents(studentList) 
+		let activeRegistrations = await registrationFile.activeRegistrations()
 
-		activeStudents.sort( (a, b) => a.first_name.localeCompare(b.first_name, 'fr', {
+		activeStudents.sort( (a, b) => a.first_name.localeCompare(b.first_name, 'en', {
 			ignorePunctuation: true
 		}));
 
 		let renderData = {
 			path: 'none',
-			students: activeStudents
+			students: activeStudents,
+			registrations: registrationFile.activeRegistrations(),
+			titles: await module.exports.getProgramTitles(activeStudents, activeRegistrations),
 		}
 		
-		response.render('index', renderData);
+		response.render('index', renderData)
 	},
 
 	filter: async function (request, response) {
 		let studentList = await db.getStudentsList();
 		let filteredGrade = request.params.grade;
 
-		let activeStudents = [];
-		for (let i = 0; i < studentList.length; i++) {
-			if (studentList[i].status == "active") {
-				activeStudents.push(studentList[i]);
-			}
-		}
+		let activeStudents = programFile.activeStudents(studentList) 
 
-		let filteredStudents = [];
+		let filteredStudents = []
 		for (let i = 0; i < activeStudents.length; i++) {
 			if (activeStudents[i].grade == filteredGrade) {
-				filteredStudents.push(activeStudents[i]);
+				filteredStudents.push(activeStudents[i])
 			}
 		}
+		let activeRegistrations = await registrationFile.activeRegistrations()
 
-		filteredStudents.sort( (a, b) => a.first_name.localeCompare(b.first_name, 'fr', {
+		filteredStudents.sort( (a, b) => a.first_name.localeCompare(b.first_name, 'en', {
 			ignorePunctuation: true
 		}));
 
 		let renderData = {
 			path: filteredGrade,
-			students: filteredStudents
+			students: filteredStudents,
+			registrations: registrationFile.activeRegistrations(),
+			titles: await module.exports.getProgramTitles(activeStudents, activeRegistrations),
 		}
 
-		response.render('index', renderData);
-	}
+		response.render('index', renderData)
+	},
+	getProgramTitles: async function(activeStudents, activeRegistrations) {
+		let programTitles = []
+		for (let i = 0; i < activeStudents.length; i++) {
+			for (let j = 0; j < activeRegistrations.length; j++) { 
+				if (activeRegistrations[j].student == activeStudents[i].id) { 
```

Comment:
this is fine, but again it could probably be accomplished using a mongoose query

Reply:
24 is the length of the id string.